pkg/framework/transportApi.go

```diff
@@ -0,0 +1,25 @@
+package framework
```

Comment:
This is not a framework... is more a utility so I suggest to rename the package or relocate the utility in other existing package or into an existing collection of utilities.

Reply:
As discussed offline, we will do this change in another PR